File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:

    vector<int>visited;

    void dfs(int node,vector<vector<int>>&graph){
        int n=graph.size();
        visited[node]=1;

        for(int i=0;i<n;i++){
            if(graph[node][i]==1 && !visited[i]){
                dfs(i,graph);
            }
        }
    }

    int findCircleNum(vector<vector<int>>& isConnected) {
        int n=isConnected.size();
        int ans=0;
        visited.resize(n,0);
        for(int i=0;i<n;i++){
            if(!visited[i]){
                ans++;
                dfs(i,isConnected);
            }
        }
        return ans;
    }
};

File: 0547-number-of-provinces/0547-number-of-provinces.cs
public class Solution {

    int[] visited;

    public void dfs(int node,int[][] isConnected){
        visited[node]=1;
        int n=isConnected.Length;

        for(int i=0;i<n;i++){
            if(isConnected[node][i]==1 && visited[i]==0){
                visited[i]=1;
                dfs(i,isConnected);
            }
        }
    }

    public int FindCircleNum(int[][] isConnected) {
        int ans=0;
        int n=isConnected.Length;
        visited=new int[n];

        for(int i=0;i<n;i++){
            if(visited[i]==0){
                dfs(i,isConnected);
                ans++;
            }
        }
        return ans;
    }
}